profiling/src/string_table/bump_owned.rs

```diff
@@ -0,0 +1,70 @@
+use bumpalo::{collections, Bump};
+use std::ops::Range;
+
+pub struct BorrowedStringTable<'b> {
+    arena: &'b Bump,
```

Comment:
Interesting pattern, this comment is mostly trying to understand more.
Why is it better to do this (take a ref to an existing arena guaranteeing the lifetime (using `b`) rather than just have the arena within the object ?

Reply:
Rust doesn't have a great self-referential story yet. That's why I built 3 structs (borrowed, bump-owned, and owned) so that this is done in safe Rust and the risky bits are in a well tested crate.

Maybe it can be done in a simpler way, but I didn't see it at the time.